Tests/LibC/TestWchar.cpp

```diff
@@ -43,3 +43,31 @@ TEST_CASE(wcscoll)
     EXPECT(wcscoll(L"a", L"z") < 0);
     EXPECT(wcscoll(L"z", L"a") > 0);
 }
+
+TEST_CASE(mbsinit)
+{
+    // Ensure that nullptr is considered an initial state.
+    EXPECT(mbsinit(nullptr) != 0);
+
+    // Ensure that a zero-initialized state is recognized as initial state.
+    mbstate_t state = {};
+    EXPECT(mbsinit(&state) != 0);
+
+    // Read a partial multibyte sequence (0b11011111 / 0xdf).
+    size_t ret = mbrtowc(nullptr, "\xdf", 1, &state);
+
+    if (ret != -2ul)
+        FAIL(String::formatted("mbrtowc accepted partial multibyte sequence with return code {} (expected -2)", static_cast<ssize_t>(ret)));
```

Comment:
`(size_t) -1` is another possible return value (hooray for POSIX), so I wanted to display the actual return value as a signed value instead of unsigned (since it's more unlikely that we will ever consume enough bytes to reach that value the other way around, at least for this test case).